SocketRocket/Internal/Security/SRSecurityPolicyBuilder.m

```diff
@@ -0,0 +1,213 @@
+
+    // Validate certificate chain for the stream.
+    NSDictionary<NSString *, id> *sslOptions = @{ (__bridge NSString *)kCFStreamSSLValidatesCertificateChain : @(YES) };
+    [stream setProperty:sslOptions forKey:(__bridge NSString *)kCFStreamPropertySSLSettings];
+}
+
+- (BOOL)evaluateServerTrust:(SecTrustRef)serverTrust
+                  forDomain:(NSString*)domain
+{
+    // Certificate chain verification is configured at the stream level, but no
+    // further evaluation happens in this policy.
+    return YES;
+}
+
+@end
+
+
+/**
+ `SRWithoutSSLValidationSecurityPolicy` requires encryption, but does no verification
+ of certificate authenticity. If you are using certificates not signed by a
+ recognized certificate authority (like a self signed certificate) consider using
+ `SRPinningSecurityPolicy` instead.
+ */
+@interface SRWithoutSSLValidationSecurityPolicy : NSObject <SRSecurityPolicy>
+
+@end
+
+@implementation SRWithoutSSLValidationSecurityPolicy
+
+- (void)updateSecurityOptionsInStream:(NSStream *)stream
+{
+    // Enforce TLS 1.2
+    [stream setProperty:(__bridge id)CFSTR("kCFStreamSocketSecurityLevelTLSv1_2") forKey:(__bridge id)kCFStreamPropertySocketSecurityLevel];
+
+    // Do not validate certificate chain for the stream.
+    NSDictionary<NSString *, id> *sslOptions = @{ (__bridge NSString *)kCFStreamSSLValidatesCertificateChain : @(NO) };
+    [stream setProperty:sslOptions forKey:(__bridge NSString *)kCFStreamPropertySSLSettings];
+}
+
+- (BOOL)evaluateServerTrust:(SecTrustRef)serverTrust
+                  forDomain:(NSString*)domain
+{
+    // no verification
+    return YES;
+}
+
+@end
+
+
+/**
+ `SRNoSecurityPolicy` is not suitable for anything except local debugging. This
+ policy specifies that no encryption is used, and thus, no verification occurs.
+ */
+@interface SRUnencryptedPolicy : NSObject <SRSecurityPolicy>
+
+@end
+
+@implementation SRUnencryptedPolicy
+
+- (void)updateSecurityOptionsInStream:(NSStream *)stream
+{
+    // no-op, no encryption.
+}
+
+- (BOOL)evaluateServerTrust:(SecTrustRef)serverTrust
+                  forDomain:(NSString*)domain
+{
+    // no verification
+    return YES;
+}
+
+@end
+
+
+@implementation SRSecurityPolicyBuilder
+
++ (SRPinningSecurityPolicy *)pinnningPolicyWithCertificates:(NSArray *)pinnedCertificates
+{
+    return [[SRPinningSecurityPolicy alloc] initWithPinnedCertificates:pinnedCertificates];
+}
+
++ (SRDefaultSecurityPolicy *)defaultPolicy
+{
+    return [SRDefaultSecurityPolicy new];
+}
+
++ (SRUnencryptedPolicy *)unencryptedPolicy
+{
+    return [SRUnencryptedPolicy new];
+}
+
++ (SRWithoutSSLValidationSecurityPolicy *)withoutSSLValidationPolicy
+{
+    return [SRWithoutSSLValidationSecurityPolicy new];
+}
+
++ (id<SRSecurityPolicy>)createWithChainValidationEnabled:(BOOL)allowsUntrustedSSLCertificates
+                                      pinnedCertificates:(nullable NSArray *)pinnedCertificates
+{
+    if (pinnedCertificates)
+    {
```

Comment:
Nit: Please use common code style that we have aka braces are on the same line as `if` or `else`.